src/test/consensus/Validations_test.cpp

```diff
@@ -257,238 +291,299 @@ class Validations_test : public beast::unit_test::suite
-                BEAST_EXPECT(harness.vals().numTrustedForLedger(Ledger::ID{2}) == 0);
-                BEAST_EXPECT(harness.vals().numTrustedForLedger(Ledger::ID{20}) == 1);
-                {
-                    // Should be the only trusted for ID{20}
-                    auto trustedVals =
-                        harness.vals().getTrustedForLedger(Ledger::ID{20});
-                    BEAST_EXPECT(trustedVals.size() == 1);
-                    BEAST_EXPECT(trustedVals[0].key() == a.currKey());
-                    // ... and should be the only node after ID{2}
-                    BEAST_EXPECT(harness.vals().getNodesAfter(Ledger::ID{2}) == 1);
+            // Cannot re-do the same full validation sequence
+            BEAST_EXPECT(
+                ValStatus::badSeq == harness.add(n.validate(ledgerAB)));
+            // Cannot send the same partial validation sequence
+            BEAST_EXPECT(
+                ValStatus::badSeq == harness.add(n.partial(ledgerAB)));
 
-                }
+            // Now trusts the newest ledger too
+            harness.clock().advance(1s);
+            BEAST_EXPECT(
+                ValStatus::current == harness.add(n.validate(ledgerABC)));
+            BEAST_EXPECT(
+                harness.vals().numTrustedForLedger(ledgerAB.id()) == 1);
+            BEAST_EXPECT(
+                harness.vals().numTrustedForLedger(ledgerABC.id()) == 1);
 
-                // A new key, but re-issue a validation with the same ID and
-                // Sequence
-                a.advanceKey();
+            // Processing validations out of order should ignore the older
+            // validation
+            harness.clock().advance(2s);
+            auto const val3 = n.validate(ledgerABCDE);
 
-                BEAST_EXPECT(
-                    AddOutcome::sameSeq == harness.add(a, Ledger::Seq{2}, Ledger::ID{20}));
-                {
-                    // Still the only trusted validation for ID{20}
-                    auto trustedVals =
-                        harness.vals().getTrustedForLedger(Ledger::ID{20});
-                    BEAST_EXPECT(trustedVals.size() == 1);
-                    BEAST_EXPECT(trustedVals[0].key() == a.currKey());
-                    // and still follows ID{2} since it was a re-issue
-                    BEAST_EXPECT(harness.vals().getNodesAfter(Ledger::ID{2}) == 1);
-                }
-            }
+            harness.clock().advance(4s);
+            auto const val4 = n.validate(ledgerABCD);
 
-            {
-                // Processing validations out of order should ignore the older
-                harness.clock().advance(2s);
-                auto const val3 = a.validation(Ledger::Seq{3}, Ledger::ID{3});
+            BEAST_EXPECT(ValStatus::current == harness.add(val4));
 
-                harness.clock().advance(4s);
-                auto const val4 = a.validation(Ledger::Seq{4}, Ledger::ID{4});
+            BEAST_EXPECT(ValStatus::stale == harness.add(val3));
```

Comment:
I realize this isn't different than before, but why should we care if a validation has a higher sequence but an earlier `signTime` than a previously seen validation from the same signer?

Reply:
Good question. I think a node that signs out of order like this is already faulty, as I don't see how an node follow the protocol would issue these validations. In that case, if the node operator quickly fixed and restarted the node, we'd rather rely on the most recent validation than the earlier one.